src/store/modules/wallet/ethWallet.ts

```diff
@@ -58,8 +59,19 @@ export class EthereumModule extends VuexModule.With({
 
   @action async connect() {
     try {
+      sendGTMEvent("Wallet Connect Select a Wallet Popup", "Wallet Connect");
       await onboard.walletSelect();
+      const state = onboard.getState();
+      sendGTMEvent("Wallet Connect Wallet Icon Click", "Wallet Connect", {
+        wallet_name: state.wallet.name
+      });
       await onboard.walletCheck();
+      sendWalletEvent(
+        "Wallet Connect",
+        "Wallet Connect",
+        state.address,
+        state.wallet.name ?? ""
+      );
```

Comment:
Should only need to send the ```userAddress``` and ```walletName``` params and ```walletName``` is optional so need not even be passed at all as it is here.